utilities/virt.py

```
@@ -1794,7 +1794,18 @@ def wait_for_running_vm(
         if check_ssh_connectivity:
             wait_for_ssh_connectivity(vm=vm, timeout=ssh_timeout)
     except TimeoutExpiredError:
-        collect_vnc_screenshot_for_vms(vm_name=vm.name, vm_namespace=vm.namespace)  # type: ignore[arg-type]
+        printable_status = vm.instance.get("status", {}).get("printableStatus")
+        if printable_status in VM_ERROR_STATUSES:
+            LOGGER.error(
+                f"VM {vm.name} timed out in '{printable_status}' after {wait_until_running_timeout}s. "
+                f"Skipping VNC screenshot as VM never booted."
+            )
```

Ruff (0.15.2)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)